src/Microsoft.Azure.WebJobs.Host/TypeUtility.cs

```diff
@@ -98,6 +98,25 @@ internal static Type UnwrapTaskType(Type type)
             return null;
         }
 
+        internal static TAttribute GetResolvedAttribute<TAttribute>(ParameterInfo parameter) where TAttribute : Attribute
+        {
+            var attribute = parameter.GetCustomAttribute<TAttribute>();
+
+            var storageAttribute = attribute as StorageAttribute;
```

Comment:
Yes, this is the main difference. I strongly prefer this to the other because it keeps this issue internal - doesn't introduce a new "hierarchical attribute merging" feature that customers can start depending on. This proposal solves all our feature needs w/o introducing any new feature surface area.